cephfs/path_test.go

```diff
@@ -60,3 +60,273 @@ func TestRemoveDir(t *testing.T) {
+			assert.NoError(t, f1.Close())
+			assert.NoError(t, mount.Unlink(fname))
+		}()
+		err = mount.Link(fname, dirname)
+		// Error, destination exists.
+		assert.Error(t, err)
+	})
+
+	t.Run("file operations", func(t *testing.T) {
+		fname := "notExist.txt"
+		err := mount.Link(fname, "hardlnk")
+		// Error, file does not exist.
+		assert.Error(t, err)
+
+		// Hard link creation.
+		fname1 := "TestFile1.txt"
+		f1, err := mount.Open(fname1, os.O_WRONLY|os.O_CREATE, 0666)
+		assert.NotNil(t, f1)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, f1.Close())
+			assert.NoError(t, mount.Unlink(fname1))
+		}()
+		err = mount.Link(fname1, "hardlnk")
+		defer func() { assert.NoError(t, mount.Unlink("hardlnk")) }()
+		// No error, normal link operation.
+		assert.NoError(t, err)
+		// Verify that link got created.
+		_, err = os.Stat(path.Join(CephMountDir, "hardlnk"))
+		assert.NoError(t, err)
+
+		// Create hard link when destination exists.
+		fname2 := "TestFile2.txt"
+		f2, err := mount.Open(fname2, os.O_WRONLY|os.O_CREATE, 0666)
+		assert.NotNil(t, f2)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, f2.Close())
+			assert.NoError(t, mount.Unlink(fname2))
+		}()
+		err = mount.Link(fname1, fname2)
+		// Error, destination already exists.
+		assert.Error(t, err)
+	})
+}
+
+func TestUnlink(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("file unlink", func(t *testing.T) {
+		fname := "TestFile.txt"
+		err := mount.Unlink(fname)
+		// Error, file does not exist.
+		assert.Error(t, err)
+
+		f, err := mount.Open(fname, os.O_WRONLY|os.O_CREATE, 0666)
+		assert.NotNil(t, f)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, f.Close())
+			assert.NoError(t, mount.Unlink(fname))
+		}()
+		assert.NoError(t, mount.Link(fname, "hardlnk"))
+
+		err = mount.Unlink("hardlnk")
+		// No Error, link will be removed.
+		assert.NoError(t, err)
+	})
+
+	t.Run("directory unlink", func(t *testing.T) {
+		dirname := "/a"
+		err := mount.MakeDir(dirname, 0755)
+		assert.NoError(t, err)
+		defer func() {
+			assert.NoError(t, mount.RemoveDir(dirname))
+		}()
+
+		err = mount.Unlink(dirname)
+		// Error, not permitted on directory.
+		assert.Error(t, err)
+	})
+}
+
+func TestSymlink(t *testing.T) {
+	mount := fsConnect(t)
+	defer fsDisconnect(t, mount)
+
+	t.Run("file operations - source doesn't exist", func(t *testing.T) {
+		fname1 := "TestFile1.txt"
+
+		err := mount.Symlink(fname1, "Symlnk1")
+		// No Error, symlink works even if source file doesn't exist.
+		assert.NoError(t, err)
+		_, err = os.Stat(path.Join(CephMountDir, "Symlnk1"))
+		// Error, source is not there.
+		assert.Error(t, err)
+
+		localPath := path.Join(CephMountDir, fname1)
+		_, err = os.Stat(localPath)
+		// Error, source file is still not there.
+		assert.EqualError(t, err,
+			fmt.Sprintf("stat %s: no such file or directory", localPath))
+
+		f1, err := mount.Open(fname1, os.O_RDWR|os.O_CREATE, 0666)
```

Comment:
Similar to above comments this seperate conflict case sub-section ought to be its own sub-test.

Reply:
Done